pynxtools/dataconverter/readers/stm/reader.py

```diff
@@ -0,0 +1,228 @@
+"""
+    A short description on SPM reader.
+"""
+
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+from typing import Any, Dict, Tuple, Union
+from collections.abc import Callable
+import json
+import yaml
+
+from pynxtools.dataconverter.readers.base.reader import BaseReader
+from pynxtools.dataconverter.template import Template
+from pynxtools.dataconverter.readers.stm.sts_file_parser import from_dat_file_into_template
```

Comment:
I see STS seems some dat file format while STM is nanonis, is the dat file generated from a different instrument?
I am asking this as I think it is useful to document in the source code that even though these are just two examples they follow the concept for users,
hej you are a scientist working in STM/STS, we know there are many different formats and different instruments: here is one common concept exemplified for these specific two types of combinations (machine A + file format A) and (machine B + file format B) that sort of story